docs/sandwich-tests/index.md

```diff
@@ -0,0 +1,383 @@
+chatroom](https://melange.re/v3.0.0/community.html#community), you get a
+recommendation from user MonadicFanatic1984 to try out
+[melange-fest](https://github.com/ahrefs/melange-fest), a library that allows
+you to write tests in OCaml and run them in [Node test
+runner](https://nodejs.org/api/test.html#test-runner). You decide to give it a
+shot!
+
+Run this from your project's root directory:
+
+```shell
+opam install melange-fest
+```
+
+Once that finishes, run `opam list` to see all the opam packages you've
+installed in your *opam switch* (more on that later). Scroll down to the entry
+for `melange-fest` and you should see something like this:
+
+```
+melange-fest    0.1.0   A minimal test framework for Melange using Node test runner
+```
+
+Open `melange-for-react-devs.opam` and add a corresponding entry in the
+`depends` section for `melange-fest`:
+
+```
+depends: [
+  "ocaml" {>= "5.1.1"}
+  "reason" {>= "3.10.0"}
+  "dune" {>= "3.8"}
+  "melange" {>= "3.0.0-51"}
+  "reason-react" {>= "0.14.0"}
+  "reason-react-ppx" {>= "0.14.0"}
+  "melange-fest" {>= "0.1.0"} // [!code ++]
+  "opam-check-npm-deps" {with-test} # todo: use with-dev-setup once opam 2.2 is out
+  "ocaml-lsp-server" {with-test} # todo: use with-dev-setup once opam 2.2 is out
+  "dot-merlin-reader" {with-test} # todo: use with-dev-setup once opam 2.2 is out
+  "odoc" {with-doc}
+]
+```
+
+Note that the version number might not be `0.1.0` when you run `opam list`. If
+it's different, just use that version number instead.
+
+Now if we want to install this project on another computer, we don't need to
+manually install `melange-fest`; it will be installed along with all the other
+dependencies when we run `opam install . --deps-only`[^1] (this is already done
+for you when you run `npm run init`).
+
+## Opam switch
+
+An [opam switch](https://ocaml.org/docs/opam-switch-introduction) is an isolated
+OCaml environment. In this book, we only use [local
+switches](https://opam.ocaml.org/blog/opam-local-switches/), which are similar
+to Node project directories[^2]. You can list all the opam switches on your
+computer by running
+
+```shell
+opam switch
+```
+
+The output will look something like this:
+
+```
+#  switch                     compiler                   description
+→  ~/melange-for-react-devs   ocaml-base-compiler.5.1.1  ~/melange-for-react-devs
+   default                    ocaml.5.1.0                default
+
+[NOTE] Current switch has been selected based on the current directory.
+       The current global system switch is default.
+```
+
+As implied by `[NOTE]`, you don't need to manually set the opam switch for your
+project, the switch is set based on your current working directory[^3].
+
+## First test
+
+Add a new file `src/order-confirmation/SandwichTests.re` and add a simple test
+to it:
+
+<<< SandwichTests.re#first-test
+
+You should get this error:
+
+```
+File "docs/burger-tests/SandwichTests.re", line 2, characters 0-9:
```

Comment:
Above it says `order-confirmation` folder, but here it says `burger-test`.

Reply:
Fixed in https://github.com/melange-re/melange-for-react-devs/pull/24/commits/357eb291c13fbda4d4e65644c0d379b9e5bc71c5